File: Recursion/Cpp/Practice/power.cpp
#include <iostream>
using namespace std;

int power(int x, int y) {
    if(y == 0) {
        return 1;
    }
    int pow = power(x,y-1);
    int ans = x*pow;
    return ans;
}

int main() {
    int x = 4;
    int y = 3; 
    cout << power(x,y);
    
}

File: Python/Recursion/power.py
def pow(x, n):
    if n == 0:
        return 1
    return x * pow(x, n-1)

def fastpow(x, n):
    if n == 0:
        return 1
    if n == 1:
        return x
    if n%2 == 1:
        return x * pow(x, n/2) * pow(x, n/2)
    return pow(x, n/2) * pow(x, n/2)

print(fastpow(2,4))
print(pow(2,4))

